test/app-tap/datetime.test.lua

```diff
@@ -1316,6 +1319,29 @@ test:test("Time intervals creation - range checks", function(test)
         local err_msg, attribs = unpack(row)
         assert_raises(test, err_msg, function() return inew(attribs) end)
     end
+
+    local boundary_values = {
+        {{ year = -MAX_YEAR_RANGE }, '-11759221 years'},
+        {{ year =  MAX_YEAR_RANGE }, '+11759221 years'},
+        {{ month = -MAX_MONTH_RANGE }, '-141110652 months'},
+        {{ month =  MAX_MONTH_RANGE }, '+141110652 months'},
+        {{ week = -MAX_WEEK_RANGE }, '-613579352 weeks'},
+        {{ week =  MAX_WEEK_RANGE }, '+613579352 weeks'},
+        {{ day = -math.floor(MAX_DAY_RANGE) }, '-4295055470 days'},
+        {{ day =  math.floor(MAX_DAY_RANGE) }, '+4295055470 days'},
+        {{ hour = -MAX_HOUR_RANGE }, '-103081331286 hours'},
+        {{ hour =  MAX_HOUR_RANGE }, '+103081331286 hours'},
+        {{ min = -MAX_MIN_RANGE }, '-6184879877160 minutes'},
+        {{ min =  MAX_MIN_RANGE }, '+6184879877160 minutes'},
+        {{ sec = -MAX_SEC_RANGE }, '-371092792629600 seconds'},
+        {{ sec =  MAX_SEC_RANGE }, '+371092792629600 seconds'},
+        {{ nsec = -MAX_NSEC_RANGE }, '+0 seconds, -2147483647 nanoseconds'},
+        {{ nsec =  MAX_NSEC_RANGE }, '+0 seconds, 2147483647 nanoseconds'},
+    }
+    for _, row in pairs(boundary_values) do
+        local args, str = unpack(row)
+        test:is(tostring(inew(args)), str)
```

Comment:
Nit:
`%s/str/expected`
`%s/args/value`
The current phrasing is a bit less intuitive, but feel free to ignore.

Reply:
To check that this is actually a boundary value, we can expect error for `+-1` values (with `assert_raises()`). I think it would be nice.

Reply:
Thanks, fixed both